google/cloud/storage/internal/credential_constants.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_CREDENTIAL_CONSTANTS_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_CREDENTIAL_CONSTANTS_H_
+
+#include "google/cloud/storage/version.h"
+#include <chrono>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+
+/**
+ * Supported signing algorithms used in JWT auth flows.
+ *
+ * We currently only support RSA with SHA-256, but use this enum for
+ * readability and easy addition of support for other algorithms.
+ */
+enum class JwtSigningAlgorithms { RS256 };
+
+/// The endpoint to create an access token from.
+inline const char* GoogleOAuthRefreshEndpoint() {
```

Comment:
Elsewhere in `google-cloud-cpp` we have been using `char const*` (note the position of the `const` qualifier).